buildpacks/ruby/src/layers/ruby_install_layer.rs

```diff
@@ -174,6 +194,9 @@ pub(crate) enum RubyInstallError {
     #[error("Invalid base url {0}")]
     InvalidBaseUrl(String),
 
+    #[error("Unsupported architecture: {0}")]
```

Comment:
I think we want to go a little above and beyond and list out the supported architectures. 

In general, any time we're failing because a value was unexpected based on a list of currently known/supported values, I want to output both what was actually output (the current architecture) but also what was expected (either arm64 or amd64 etc.)

This can help the future development errors/issues for example if someone accidentally adds a space like `" arm64"`. Even if we think such an issue would be rare, I don't see a downside in adding that information to the error message (other than that requirement will influence the implementation, such as needing to move the information to a const, for example.